perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3327,434 +3338,488 @@ private boolean isThisLightweightSynchronization(PerunSession sess, Group group)
+						// if update fails, skip him
+						log.warn("Can't update member from candidate {} due to attribute value exception {}.", candidate, e);
+						skippedMembers.add("MemberEntry:[" + candidate + "] was skipped because there was problem when updating member from candidate: Exception: " + e.getName() + " => '" + e2.getMessage() + "'");
+						return;
 					}
-				} catch (AttributeValueException e1) {
-					log.warn("Can't create member from candidate {} due to attribute value exception {}.", candidate, e1);
-					skippedMembers.add("MemberEntry:[" + candidate + "] was skipped because there was problem when createing member from candidate: Exception: " + e1.getName() + " => '" + e1.getMessage() + "'");
-					continue;
-				} catch (ExtendMembershipException e1) {
-					log.warn("Can't create member from candidate {} due to membership expiration exception {}.", candidate, e1);
-					skippedMembers.add("MemberEntry:[" + candidate + "] was skipped because membership expiration: Exception: " + e1.getName() + " => " + e1.getMessage() + "]");
-					continue;
+				} catch (Exception e2) {
+					//Something is still wrong, thrown consistency exception
+					throw new ConsistencyErrorException("Trying to add existing member (it is not possible to get him by userExtSource even if is also not possible to create him in DB)!");
 				}
+			} catch (AttributeValueException e1) {
+				log.warn("Can't create member from candidate {} due to attribute value exception {}.", candidate, e1);
+				skippedMembers.add("MemberEntry:[" + candidate + "] was skipped because there was problem when createing member from candidate: Exception: " + e1.getName() + " => '" + e1.getMessage() + "'");
+				return;
+			} catch (ExtendMembershipException e1) {
+				log.warn("Can't create member from candidate {} due to membership expiration exception {}.", candidate, e1);
+				skippedMembers.add("MemberEntry:[" + candidate + "] was skipped because membership expiration: Exception: " + e1.getName() + " => " + e1.getMessage() + "]");
+				return;
 			}
+		}
 
-			try {
-				// Add the member to the group
-				if (!group.getName().equals(VosManager.MEMBERS_GROUP)) {
-					// Do not add members to the generic members group
-					try {
-						getPerunBl().getGroupsManagerBl().addMember(sess, group, member);
-					} catch(GroupNotExistsException ex) {
-						// Shouldn't happen, because every group has at least Members group as a parent
-						// Shouldn't happen, group should always exist
-						throw new ConsistencyErrorException(ex);
-					}
+		try {
+			// Add the member to the group
+			if (!group.getName().equals(VosManager.MEMBERS_GROUP)) {
+				// Do not add members to the generic members group
+				try {
+					getPerunBl().getGroupsManagerBl().addMember(sess, group, member);
+				} catch(GroupNotExistsException ex) {
+					// Shouldn't happen, because every group has at least Members group as a parent
```

Comment:
Not part of your changes, but I believe this comment makes no sense, since groups can be top-level and don't have "members" as a parent.